src/testing/fast_array_utils/__init__.py

```diff
@@ -37,48 +40,245 @@ def __call__(  # noqa: D102
+            fn = self.to_dask_array
+        elif self.cls is types.H5Dataset:
+            fn = self.to_h5py_dataset
+        elif self.cls is types.ZarrArray:
+            fn = self.to_zarr_array
+        elif self.cls is types.CupyArray:
+            import cupy as cu
+
+            fn = cu.asarray
+        else:
+            fn = self.cls  # type: ignore[assignment]
+
+        return fn(x, dtype=dtype)
+
+    def to_dask_array(self, x: ArrayLike, /, *, dtype: DTypeLike | None = None) -> types.DaskArray:
+        """Convert to a dask array."""
+        if TYPE_CHECKING:
+            import dask.array.core as da
+        else:
+            import dask.array as da
+
+        assert self.inner is not None
+
+        arr = self.inner(x, dtype=dtype)
+        return da.from_array(arr, _half_chunk_size(arr.shape))  # type: ignore[no-untyped-call,no-any-return]
+
+    def to_h5py_dataset(
+        self, x: ArrayLike, /, *, dtype: DTypeLike | None = None
+    ) -> types.H5Dataset:
+        """Convert to a h5py dataset."""
+        if (ctx := self.conversion_context) is None:
+            msg = "`conversion_context` must be set for h5py"
+            raise RuntimeError(msg)
+        arr = np.asarray(x, dtype=dtype)
+        return ctx.hdf5_file.create_dataset(ctx.get_ds_name(), arr.shape, arr.dtype, data=arr)
+
+    @staticmethod
+    def to_zarr_array(x: ArrayLike, /, *, dtype: DTypeLike | None = None) -> types.ZarrArray:
+        """Convert to a zarr array."""
+        import zarr
+
+        arr = np.asarray(x, dtype=dtype)
+        za = zarr.create_array({}, shape=arr.shape, dtype=arr.dtype)
+        za[...] = arr
+        return za
+
+
+_SUPPORTED_TYPE_NAMES_DISK = [
+    "h5py.Dataset",
+    "zarr.Array",
+]
+_SUPPORTED_TYPE_NAMES_DENSE = [
+    "numpy.ndarray",
+    "cupy.ndarray",
+]
+_SUPPORTED_TYPE_NAMES_SPARSE = [
+    "scipy.sparse.csr_array",
+    "scipy.sparse.csc_array",
+    "scipy.sparse.csr_matrix",
+    "scipy.sparse.csc_matrix",
+    "cupyx.scipy.sparse.csr_matrix",
+    "cupyx.scipy.sparse.csc_matrix",
+]
+
+SUPPORTED_TYPES_DISK: tuple[ArrayType, ...] = tuple(
+    map(ArrayType.from_qualname, _SUPPORTED_TYPE_NAMES_DISK)
+)
+"""Supported array types that represent on-disk data
+
+These on-disk array types are not supported inside dask arrays.
+"""
+
+SUPPORTED_TYPES_MEM_DENSE: tuple[ArrayType, ...] = tuple(
+    map(ArrayType.from_qualname, _SUPPORTED_TYPE_NAMES_DENSE)
+)
+"""Supported dense in-memory array types."""
+
+SUPPORTED_TYPES_MEM_SPARSE: tuple[ArrayType, ...] = tuple(
+    map(ArrayType.from_qualname, _SUPPORTED_TYPE_NAMES_SPARSE)
+)
+"""Supported sparse in-memory array types."""
+
+SUPPORTED_TYPES_MEM: tuple[ArrayType, ...] = (
+    *SUPPORTED_TYPES_MEM_DENSE,
+    *SUPPORTED_TYPES_MEM_SPARSE,
+)
+"""Supported array types that are valid inside dask arrays."""
+
+SUPPORTED_TYPES_DASK: tuple[ArrayType, ...] = tuple(
+    ArrayType("dask.array", ".Array", t) for t in SUPPORTED_TYPES_MEM
```

Comment:
Why not `from_qualname` here?

Reply:
I could do `from_qualname("dask.array.Array", str(t))`, but that feels dirty